packages/main/src/AINotice.ts

```diff
@@ -0,0 +1,175 @@
+ * - `showIcon`: A boolean indicating whether the icon should be shown.
+ *
+ * **Note**: The layout dynamically adjusts based on the available space and the content's width. If space is limited, verification text may be hidden to maintain the layout.
+ *
+ * ### ES6 Module Import
+ * `import "@ui5/webcomponents/dist/AINotice.js";`
+ *
+ * @constructor
+ * @extends UI5Element
+ * @public
+ * @since 2.4.0
+ */
+
+@customElement({
+	tag: "ui5-ainotice",
+	renderer: litRender,
+	template: AINoticeTemplate,
+	styles: AINoticeCss,
+	dependencies: [Link, Button, Popover, Icon],
+})
+
+class AINotice extends UI5Element {
+	/**
+	 * A callback function used for resizing logic.
+	 * @private
+	 */
+	_handleResize: ResizeObserverCallback;
+
+	/**
+	 * Defines the attribution text to be displayed in the component.
+	 * @public
+	 */
+	@property()
+	attributionText = "";
+
+	/**
+	 * Defines the verification text to be displayed in the component.
+	 * @public
+	 */
+	@property()
+	verificationText = "";
+
+	/**
+	 * Defines whether an icon should be shown in the component.
+	 * @default false
+	 * @public
+	 */
+	@property({ type: Boolean })
+	showIcon = false;
+
+	/**
+	 * The popup slot allows for custom popups to be inserted into the component.
+	 * @public
+	 */
+	@slot({ type: HTMLElement })
+	popup!: HTMLElement;
+
+	@i18n("@ui5/webcomponents")
+	static i18nBundle: I18nBundle;
+
+	constructor() {
+		super();
+		this._handleResize = this.resizeWidth.bind(this);
+	}
+
+	onEnterDOM() {
+		ResizeHandler.register(this, this._handleResize);
+	}
+
+	onExitDOM() {
+		ResizeHandler.deregister(this, this._handleResize);
+	}
+
+	/**
+	 * A getter that determines whether the component is in the "emphasized" state, which includes attribution text, verification text, and an icon.
+	 */
+	get isEmphasized(): boolean {
+		return !!this.attributionText && !!this.verificationText && this.showIcon;
+	}
+
+	/**
+	 * A getter that determines whether the component is in the "icon-only" state, where only the icon is displayed.
+	 */
+	get isIconOnly(): boolean {
+		return !this.attributionText && !this.verificationText && this.showIcon;
+	}
+
+	/**
+	 * A getter that determines whether the component is in the "shortened" state, which shows only attribution text and hides verification text and icon.
+	 */
+	get isShortened(): boolean {
+		return !!this.attributionText && !this.verificationText && !this.showIcon;
+	}
+
+	get tooltipValue() {
+		return AINotice.i18nBundle.getText(AINOTICE_BUTTON_TOOLTIP);
+	}
+
+	/**
+	 * Resizes the component based on its content and the available space.
+	 * It adjusts the visibility of the verification text depending on the available width.
+	 */
+	resizeWidth() {
+		const container = this.getDomRef();
+		const verificationText = this.shadowRoot?.querySelector(".verificationText") as HTMLElement;
+		const attributionText = this.shadowRoot?.querySelector(".link-text") as HTMLElement;
```

Comment:
querySelector is generic, we can make use of it:

don't
this.shadowRoot?.querySelector<HTMLElement>(".link-text") as HTMLElement;

do
this.shadowRoot?.querySelector<HTMLElement>(".link-text");